### PR TITLE
unifying the raft and stand-along server's initialization

### PR DESCRIPTION
**Pull request by bobotu**
Use the new `InnerServer` interface to simplify the code in `node/main.go`. 

We can remove `RunRaftServer` later.

### CODE REVIEW

File: node/main.go [hunk 1]
```diff
@@ -111,16 +124,40 @@ func main() {
...
 	if err != nil {
 		log.Fatal(err)
 	}
-	log.Info("RegionManager closed.")
-	err = db.Close()
-	if err != nil {
-		log.Fatal(err)
+}
+
+func newRaftServerBuilder(bundle *mvcc.DBBundle, safePoint *tikv.SafePoint) server.InnerServer {
```
Comment (coocood) [inline node/main.go]: The returned value is not a builder.

File: node/main.go [hunk 2]
```diff
@@ -66,26 +67,38 @@ func main() {
...
+		innerServer = newStandAlongBuilder(bundle, safePoint)
+	}
+
+	pdClient, err := pd.NewClient(*pdAddr, "")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	builderCtx := server.Context{
+		PDClient: pdClient,
```
Comment (coocood) [inline node/main.go]: How about put pdClient as an argument directly for new server function?

File: node/main.go [hunk 3]
```diff
@@ -66,26 +67,38 @@ func main() {
...
+	}
+
+	if err := innerServer.Setup(&builderCtx); err != nil {
+		log.Fatal(err)
+	}
+
+	store := tikv.NewMVCCStore(bundle, *dbPath, safePoint, innerServer.GetDBWriter())
+	tikvServer := tikv.NewServer(innerServer.GetRegionManager(), store)
```
Comment (coocood) [inline node/main.go]: I think make the inner server as a field of `tikvServer` is better, so we can implement `TiKVServer` methods like
`func (svr *Server) Raft(tikvpb.Tikv_RaftServer) error {` with inner server.

The interface `InnerServer` can be defined in `tikv` package, the implementation of stand alone can be defined in the `tikv` package too, and `RaftInnerServer` can be defined in `raftstore` package. So we don't need the `server` package.

**Comment by bobotu** [comment]
PTAL @coocood

File: tikv/raftstore/server.go [hunk 4]
```diff
@@ -13,6 +13,122 @@ import (
...
+	snapManager     *SnapManager
+	coprocessorHost *CoprocessorHost
+	raftRouter      *RaftstoreRouter
+	batchSystem     *raftBatchSystem
+	pdWorker        *worker
+	resolveWorker   *worker
+	snapWorker      *worker
+}
+
+func (sb *RaftInnerServer) Raft(stream tikvpb.Tikv_RaftServer) error {
```
Comment (coocood) [inline tikv/raftstore/server.go]: `sb` -> `ris`

File: node/main.go [hunk 5]
```diff
@@ -66,26 +66,34 @@ func main() {
...
 	} else {
-		dbWriter = tikv.NewDBWriter(bundle, safePoint)
+		innerServer, dbWriter, regionManager = setupStandAlongInnerServer(bundle, safePoint, pdClient)
 	}
+
 	store := tikv.NewMVCCStore(bundle, *dbPath, safePoint, dbWriter)
-	tikvServer := tikv.NewServer(rm, store)
+	tikvServer := tikv.NewServer(regionManager, store, innerServer)
+
+	if err := innerServer.Start(pdClient); err != nil {
```
Comment (coocood) [inline node/main.go]: Can we `Start` in `setupRaftInnerServer`?

File: node/main.go [hunk 6]
```diff
@@ -66,26 +66,34 @@ func main() {
...
+	}
+
+	var (
+		innerServer   tikv.InnerServer
+		dbWriter      mvcc.DBWriter
+		regionManager tikv.RegionManager
+	)
 	if *raft {
-		dbWriter = raftstore.NewDBWriter(bundle, raftstore.NewDefaultConfig())
+		innerServer, dbWriter, regionManager = setupRaftInnerServer(bundle, safePoint, pdClient)
```
Comment (coocood) [inline node/main.go]: We can return `store` instead of `dbWriter` here.

**Comment by coocood** [comment]
LGTM

**Pull request merged**